ogr/ogrsf_frmts/miramon/mm_wrlayr.c

```diff
@@ -0,0 +1,7038 @@
+                   "driver (MMInitFlush())");
+        return 1;
+    }
+    pFlush->OffsetWhereToFlush = DiskOffsetWhereToFlush;
+    pFlush->CurrentOffset = 0;
+    return 0;
+}
+
+// Reads data from a file into a buffer.
+int MMReadFlush(struct MM_FLUSH_INFO *pFlush)
+{
+    fseek_function(pFlush->pF, pFlush->OffsetWhereToFlush, SEEK_SET);
+    if (pFlush->nBlockSize !=
+        (GUInt64)(fread_function(pFlush->pBlockWhereToSaveOrRead, 1,
+                                 (size_t)pFlush->nBlockSize, pFlush->pF)))
+        return 1;
+    return 0;
+}
+
+// Flushes data from a buffer to a disk file.
+static int MMFlushToDisk(struct MM_FLUSH_INFO *FlushInfo)
+{
+    if (!FlushInfo->nNumBytes)
+        return 0;
+    // Just flush to the disk at the correct place.
+    fseek_function(FlushInfo->pF, FlushInfo->OffsetWhereToFlush, SEEK_SET);
+
+    if (FlushInfo->nNumBytes !=
+        (GUInt64)fwrite_function(FlushInfo->pBlockWhereToSaveOrRead, 1,
+                                 (size_t)FlushInfo->nNumBytes, FlushInfo->pF))
+        return 1;
+    FlushInfo->OffsetWhereToFlush += FlushInfo->nNumBytes;
+    FlushInfo->NTimesFlushed++;
+    FlushInfo->TotalSavedBytes += FlushInfo->nNumBytes;
+    FlushInfo->nNumBytes = 0;
+
+    return 0;
+}
+
+// Reads a block of data from a buffer in memory
+int MMReadBlockFromBuffer(struct MM_FLUSH_INFO *FlushInfo)
+{
+    if (!FlushInfo->SizeOfBlockToBeSaved)
+        return 0;
+
+    if (FlushInfo->pBlockToBeSaved)
+    {
+        memcpy(FlushInfo->pBlockToBeSaved,
+               (void *)((char *)FlushInfo->pBlockWhereToSaveOrRead +
+                        FlushInfo->CurrentOffset),
+               FlushInfo->SizeOfBlockToBeSaved);
+    }
+    FlushInfo->CurrentOffset += FlushInfo->SizeOfBlockToBeSaved;
+
+    return 0;
+}
+
+// Appends a block of data to a buffer in memory, which is
+// used for later flushing to disk.
+int MMAppendBlockToBuffer(struct MM_FLUSH_INFO *FlushInfo)
+{
+    if (FlushInfo->SizeOfBlockToBeSaved)
+    {
+        // If all the bloc itselfs does not fit to the buffer,
+        // then all the block is written directly to the disk
+        if (FlushInfo->nNumBytes == 0 &&
+            FlushInfo->SizeOfBlockToBeSaved >= FlushInfo->nBlockSize)
+        {
+            if (MMFlushToDisk(FlushInfo))
+                return 1;
+            return 0;
+        }
+
+        // There is space in FlushInfo->pBlockWhereToSaveOrRead?
+        if (FlushInfo->nNumBytes + FlushInfo->SizeOfBlockToBeSaved <=
+            FlushInfo->nBlockSize)
+        {
+            if (FlushInfo->pBlockToBeSaved)
+            {
+                memcpy((void *)((char *)FlushInfo->pBlockWhereToSaveOrRead +
+                                FlushInfo->nNumBytes),
+                       FlushInfo->pBlockToBeSaved,
+                       FlushInfo->SizeOfBlockToBeSaved);
+            }
+            else  // Add zero caracters
+            {
+                char zero_caracters[8] = "\0\0\0\0\0\0\0";
```

Comment:
your version is likely correct, but below will be more obvious there will actually be 8 zeroes:
```suggestion
                char zero_caracters[8] = {0, 0, 0, 0, 0, 0, 0, 0};
```